include/alpaka/mem/buf/Traits.hpp

```diff
@@ -88,6 +94,13 @@ namespace alpaka
         return traits::AsyncBufAlloc<TElem, Dim<TExtent>, TIdx, alpaka::Dev<TQueue>>::allocAsyncBuf(queue, extent);
     }
 
+    //! Check if the given device can allocate a stream-ordered memory buffer of the given dimensionality.
+    //!
+    //! TDev is the type of device to allocate the buffer on.
+    //! TDim is the dimensionality of the buffer to allocate.
+    template<typename TDev, typename TDim>
+    constexpr bool hasAsyncBufferSupport = traits::HasAsyncBufferSupport<TDim, TDev>::value;
```

Comment:
Sorry, I only saw this now:

```suggestion
    constexpr inline bool hasAsyncBufferSupport = traits::HasAsyncBufferSupport<TDim, TDev>::value;
```

Otherwise the compiler/linker might complain about symbol duplication if this is used in multiple translation units.

Reply:
Good point - I'm not sure why I didn't get any errors in the tests or in our code base, but adding the `inline` is definitely better.

Reply:
Actually, looks like at least some versions of clang do not like the change:
```
/home/runner/work/alpaka/alpaka/include/alpaka/mem/buf/Traits.hpp:102:27: error: no previous extern declaration for non-static variable 'hasAsyncBufSupport' [-Werror,-Wmissing-variable-declarations]
    constexpr inline bool hasAsyncBufSupport = traits::HasAsyncBufSupport<TDim, TDev>::value;
                          ^
```

Reply:
Ah yes, I've encountered this in the past, too. clang 5 and 6 are buggy. Guard the line with this:

```c++
/* TODO: Remove the following pragmas once support for clang 5 and 6 is removed. They are necessary because these /  clang versions incorrectly warn about a missing 'extern'. */
#if BOOST_COMP_CLANG
#    pragma clang diagnostic push
#    pragma clang diagnostic ignored "-Wmissing-variable-declarations"
#endif

/* Your code */

#if BOOST_COMP_CLANG
#    pragma clang diagnostic pop
#endif
```

Reply:
Looking at the clang archives, this is a bug [reported for clang 5 and 6](https://github.com/llvm/llvm-project/issues/35210) and [fixed](https://code.ornl.gov/ecpcitest/proteas/llvm-project/-/commit/742e3215bf9e9aa657d13201ccb138ab2d0090fa) in later releases.

Reply:
Ah, I see that you encountered the same problem in https://github.com/alpaka-group/alpaka/blob/4096982a4befcbd39ee52659699aaaaa704f9048/include/alpaka/core/Decay.hpp#L36-L46 , so I'll copy the same workaround.

Reply:
> Ah yes, I've encountered this in the past, too. clang 5 and 6 are buggy. Guard the line with this: ...

thanks @j-stephan, for some reason I wasn't getting notified about these comments :-/